pkg/engine/engine.go

```diff
@@ -814,38 +800,48 @@ func (e *Engine) detectChunk(ctx context.Context, data detectableChunk) {
 	defer common.Recover(ctx)
 	defer cancel()
 
-	results, err := data.detector.FromData(ctx, data.chunk.Verify, data.chunk.Data)
-	if err != nil {
-		ctx.Logger().Error(err, "error scanning chunk")
-	}
+	// To reduce the overhead of regex calls in the detector, we limit the amount of data passed to each detector.
+	// The Matches method of the DetectorMatch struct is used to extract the relevant portions of the chunk data
+	// based on the start and end positions of each match. The end position is determined by taking the minimum
+	// of the keyword position + maxMatchLength and the length of the chunk data.
+	// This optimization is based on the assumption that most secrets shouldn't exceed maxMatchLength (300)
+	// characters in length from the keyword's position.
+	// The value of maxMatchLength is a constant and can be adjusted based on further analysis and requirements.
+	matchedBytes := data.detector.Matches(data.chunk.Data)
+	for _, match := range matchedBytes {
```

Comment:
Does `mergeMatches` aide in de-duplication at all? Some of [my recent changes](https://github.com/trufflesecurity/trufflehog/blob/c7b72b98677722afac24a2bafe84bd98074129b4/pkg/detectors/alchemy/alchemy.go#L39-L42) have been around de-duplicating results in chunks to prevent making the same network calls 2/10/20 times. Admittedly, a better solution would be #2262 rather than caching matches in a given chunk.